packages/react-player/src/controls.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2023 Design Barn Inc.
+ */
+
+import { PlayerState } from 'common';
+import React, { useContext, useMemo, useState } from 'react';
+
+import { DotLottieContext } from './react-player';
+
+const AVAILABLE_BUTTONS = ['play', 'stop', 'loop'] as const;
+
+interface ControlsProps extends React.HTMLAttributes<HTMLDivElement> {
+  buttons?: Array<typeof AVAILABLE_BUTTONS[number]>;
+  show?: boolean;
+}
+
+export const Controls: React.FC<ControlsProps> = ({ buttons = AVAILABLE_BUTTONS, ...props }) => {
+  const { currentState, dotLottiePlayer, frame, loop, seeker } = useContext(DotLottieContext);
+
+  const [isLoop, setIsLoop] = useState(loop);
+  const [_prevState, setPrevState] = useState(PlayerState.Loading);
```

Comment:
if the purpose of `_prevState` is to keep track of the previous player state, i'd suggest to use `usePrevious` hook 
to properly manage the previous `currentState` automatically. 

```ts
import { useState, useEffect, useRef } from "react";
// Usage
function App() {
  // State value and setter for our example
  const [count, setCount] = useState<number>(0);
  // Get the previous value (was passed into hook on last render)
  const prevCount: number = usePrevious<number>(count);
  // Display both current and previous count value
  return (
    <div>
      <h1>
        Now: {count}, before: {prevCount}
      </h1>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </div>
  );
}
// Hook
function usePrevious<T>(value: T): T {
  // The ref object is a generic container whose current property is mutable ...
  // ... and can hold any value, similar to an instance property on a class
  const ref: any = useRef<T>();
  // Store current value in ref
  useEffect(() => {
    ref.current = value;
  }, [value]); // Only re-run if value changes
  // Return previous value (happens before update in useEffect above)
  return ref.current;
}import { useState, useEffect, useRef } from "react";
// Usage
function App() {
  // State value and setter for our example
  const [count, setCount] = useState<number>(0);
  // Get the previous value (was passed into hook on last render)
  const prevCount: number = usePrevious<number>(count);
  // Display both current and previous count value
  return (
    <div>
      <h1>
        Now: {count}, before: {prevCount}
      </h1>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </div>
  );
}
// Hook
function usePrevious<T>(value: T): T {
  // The ref object is a generic container whose current property is mutable ...
  // ... and can hold any value, similar to an instance property on a class
  const ref: any = useRef<T>();
  // Store current value in ref
  useEffect(() => {
    ref.current = value;
  }, [value]); // Only re-run if value changes
  // Return previous value (happens before update in useEffect above)
  return ref.current;
}
```

Reply:
I think for this case usePrevious hook may not be useful. This is technically not the previous state. `_prevState` here is used to identify which state player should be after user stops the manual seek.  Every user drag to seek it calls these two functions

```
      dotLottiePlayer.goToAndPlay(nextFrame, true);
      dotLottiePlayer.pause();
```
So the player state changes play to pause even though previously player is at Playing state. 

I renamed the variable to `_stateBeforeSeek`. Lemme know if you have any concerns.

Reply:
Done ✅